Text-sentiment-analysis/Sentiment_Analysis_Cleaned.py

```diff
@@ -0,0 +1,151 @@
+import numpy as np
+from sklearn.preprocessing import LabelEncoder
+from sklearn.model_selection import train_test_split
+from keras.preprocessing.text import Tokenizer
+from keras.preprocessing.sequence import pad_sequences
+from nltk.tokenize import word_tokenize
+import keras
+from sklearn.metrics import classification_report
+from sklearn.metrics import accuracy_score
+import math
+import nltk
+from nltk.corpus import RegexpTokenizer as regextoken
+
+nltk.download('wordnet')
+nltk.download('stopwords')
+nltk.download('punkt')
+nltk.download('averaged_perceptron_tagger')
+
+df = pd.read_csv("/content/drive/MyDrive/Tweets.csv")
+df.head()
+
+df.columns
+
+tweet_df = df[['text','airline_sentiment']]
+print(tweet_df.shape)
+tweet_df.head(5)
+
+tweet_df = tweet_df[tweet_df['airline_sentiment'] != 'neutral']
+tweet_df.head(10)
+
+tweet_df["airline_sentiment"].value_counts()
+
+# convert airline_seentiment to numeric
+y = tweet_df.airline_sentiment.factorize()
+y
+
+# Preprocess the text with function processtext()
+stop_words = stopwords.words("english")
+stop_words.extend(["virginamerica", "VirginAmerica", "americanair", "usairways", "jetblue", "southwestair", "@"])
+punctuations = list(string.punctuation)
+lemma = WordNetLemmatizer() # for Lemmatisation
+
+def processtext(text):
+    # text = [word for word in text.split() if word not in (stop_words)]
+    text=re.sub("[^a-zA-Z]"," ",text) # Filter to allow only alphabets in text
+    text=text.lower() # Convert the text to lowercase to maintain consistency
+    tokens=word_tokenize(text) # Tokenize the text
+    tokens=[token for token in tokens if token not in stop_words] 
+    tokens=[lemma.lemmatize(token) for token in tokens if token not in punctuations] # Lemmatisation of tokens
+    tokens=[token for token in tokens if token not in stop_words] 
+    text=" ".join(tokens)
+    return text
+
+tweet_df["processed_text"] = tweet_df.text.apply(lambda x: processtext(x))
+
+tweet_df.head()
+
+tweet = tweet_df.text.values
+tokenizer = Tokenizer(num_words=5000)
+tokenizer.fit_on_texts(tweet)
+vocab_size = len(tokenizer.word_index) + 1
+encoded_docs = tokenizer.texts_to_sequences(tweet)
+X = pad_sequences(encoded_docs, maxlen=200)
+
+# Build the model
+from tensorflow.keras.models import Sequential
+from tensorflow.keras.layers import LSTM,Dense, Dropout, SpatialDropout1D
+from tensorflow.keras.layers import Embedding
+
+embedding_vector_length = 32
+model = Sequential() 
+model.add(Embedding(vocab_size, embedding_vector_length, input_length=200) )
+model.add(SpatialDropout1D(0.25))
+model.add(LSTM(50, dropout=0.5, recurrent_dropout=0.5))
+model.add(Dropout(0.2))
+model.add(Dense(1, activation='sigmoid')) 
+model.compile(loss='binary_crossentropy',optimizer='adam', metrics=['accuracy'])  
+print(model.summary())
+
+
+
+
+history = model.fit(X, y[0],validation_split=0.2, epochs=5, batch_size=32)
+
+# reviews on which we need to predict
+sentence = ["The movie was very touching and heart whelming", 
```

Comment:
Why you have used movie review sentences here?
The dataset was for flight review tweets, right.

Reply:
Yh, it was the sentence that was used for testing in the previous notebook. I didn't see it has a problem since we were predicting sentiments. Do you think I should change it?